app.js

```diff
@@ -21,78 +21,41 @@ client.login(process.env.DISCORD_TOKEN);
 
 client.on('messageCreate', async function (message)
 {
-    if (message.content.includes('https://twitter.com/'))
-    {        
-        let name = '';
-        let guild = client.guilds.cache.get(message.guildId);
-        guild.members
-            .fetch(message.author)
-            .then(data => name = data.nickname + ' (' + data.user.username + ')');
+    const addressToChangeTwitterTo = "vxtwitter.com";
+    let messageContent = message?.content ?? "";
+    let repostTweet = true;
+
+    // switch to vxtwitter
+    switch(expression) {
+        case messageContent.includes('https://twitter.com/'):
+          messageContent = messageContent.replace('twitter.com', addressToChangeTwitterTo);
+          break;
+        case messageContent.includes('https://x.com/'):
+            messageContent = messageContent.replace('x.com', addressToChangeTwitterTo);
+          break;
+        default:
+            repostTweet = false;
+    }
+
+    // should be a !repost tweet + return statment but I can't test in this environment
+    if(repostTweet) {
 
-        let tweetId = message.content.match(/\/status\/(\d+)/)[1];
-        if (await TweetContainsVideo(tweetId))
-        {
-            message.delete();
-            message.channel.send(name + ' posted: ' + message.content.replace('twitter.com', 'vxtwitter.com'));
+        // strip tracking link
+        if (messageContent.match(/\?t=/gm) != null) {
```

Comment:
This can definitely be simplified. Something like
```suggestion
        messageContent = messageContent.match(/.+?(?=\?t=)/gm)?.[0] ?? messageContent
```

I haven't checked if the regex actually works, but I'll do that once you simplify the line

Reply:
it does and reading skill issue

Reply:
oh actually I didn't read this comment fully and this is a good idea

Reply:
no actually I lied, I like how it segregates detecting the link from stripping it. It might not be optimized but it allows for further expansion down the line.

Reply:
Question then, why are the two regexes different?

Reply:
One matches the identifier '?t=' one selects everything before it.